services/profile/service/profile_service.go

```diff
@@ -142,3 +143,85 @@ func GetAllProfiles() (*models.ProfileList, error) {
 
 	return &profile_list, nil
 }
+
+/*
+	Returns a list of "limit" profiles sorted decesending by points.
+	If "limit" is not provided, this will return a list of all profiles.
+*/
+func GetProfileLeaderboard(limit int) (*models.ProfileList, error) {
+	profiles := []models.Profile{}
+
+	sort_field := database.SortField{
+		Name:     "points",
+		Reversed: true,
+	}
+
+	err := db.FindAllSorted("profiles", nil, []database.SortField{sort_field}, &profiles)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if limit != 0 { // If no limit is provided, it will default to 0, and all profiles will be returned
+		profiles = profiles[:limit]
+	}
+
+	profile_list := models.ProfileList{
+		Profiles: profiles,
+	}
+
+	return &profile_list, nil
+}
+
+/*
+	Returns a list of profiles filtered upon teamStatus and interests. Will be limited to only include the first "limit" results.
+*/
+func GetFilteredProfiles(teamStatus string, interests_string string, limit int) (*models.ProfileList, error) {
```

Comment:
Could we move the filtering logic into mongodb as much as possible?

Our existing database interface (/common/database) should be able to handle the `teamStatus` and `interests` queries. Sorting by matching interests will be trickier and isn't supported by our database interface AFAIK.